src/Analyzers/Core/Analyzers/UseCollectionInitializer/AbstractUseCollectionInitializerAnalyzer.cs

```diff
@@ -90,233 +95,109 @@ public void Dispose()
             // However, for collection initializers we always want at least one element to add to the initializer.  In
             // other words, we don't want to suggest changing `new List<int>()` to `new List<int>() { }` as that's just
             // noise.  So convert empty results to an invalid result here.
-            if (areCollectionExpressionsSupported)
+            if (analyzeForCollectionExpression)
                 return result;
 
             // Downgrade an empty result to a failure for the normal collection-initializer case.
             return result.IsEmpty ? default : result;
         }
 
-        protected override bool TryAddMatches(ArrayBuilder<Match<TStatementSyntax>> matches)
+        protected override bool TryAddMatches(
+            ArrayBuilder<Match<TStatementSyntax>> matches, CancellationToken cancellationToken)
         {
-            // If containing statement is inside a block (e.g. method), than we need to iterate through its child statements.
-            // If containing statement is in top-level code, than we need to iterate through child statements of containing compilation unit.
-            var containingBlockOrCompilationUnit = _containingStatement.GetRequiredParent();
-
-            // In case of top-level code parent of the statement will be GlobalStatementSyntax,
-            // so we need to get its parent in order to get CompilationUnitSyntax
-            if (_syntaxFacts.IsGlobalStatement(containingBlockOrCompilationUnit))
-            {
-                containingBlockOrCompilationUnit = containingBlockOrCompilationUnit.Parent!;
-            }
-
-            var foundStatement = false;
```

Comment:
this was logic duplicated in a few places for effectively saying "i have a statement, find me the statements that follow (including in a global context).  This logic was unified into a single helper.